packit/patches.py

```diff
@@ -0,0 +1,235 @@
+                    return False, commit
+        return True, None
+
+    @staticmethod
+    def linearize_history(git_ref: str):
+        r"""
+        Transform complex git history into a linear one starting from a selected git ref.
+
+        Change this:
+        * | | | | | | | |   ea500ac513 (tag: v245) Merge pull request #15...
+        |\ \ \ \ \ \ \ \ \
+        | * | | | | | | | | 0d5aef3eb5 hwdb: update for v245
+        | | |_|_|_|_|_|/ /
+        | |/| | | | | | |
+        * | | | | | | | | 03985d069b NEWS: final contributor update for v245
+
+        Into this:
+        * 0d5aef3eb5 hwdb: update for v245
+        * 03985d069b NEWS: final contributor update for v245
+        """
+        logger.info(
+            "When git history is too complex with merge commits having parents \n"
+            "across a wide range, git is known to produce patches which cannot be applied. \n"
+            "Therefore we are going to make the history linear on a dedicated branch \n"
+            "to make sure the patches will be able to be applied."
+        )
+        current_time = datetime.datetime.now().strftime(DATETIME_FORMAT)
+        target_branch = f"packit-patches-{current_time}"
+        logger.info(f"Switch branch to {target_branch}")
+        run_command(["git", "checkout", "-B", target_branch])
+        target = f"{git_ref}..HEAD"
+        logger.debug(f"Linearize history {target}")
+        # https://stackoverflow.com/a/17994534/909579
+        # With this command we will rewrite git history of our newly created branch
+        # by dropping the merge commits and setting parent commits to those from target branch
+        # this means we will drop the reference from which we are merging
+        # filter branch passes these to cut:
+        #   ` -p 61f3e897f13101f29fb8027e8839498a469ad58e`
+        #   ` -p b7cf4b4ef5d0336443f21809b1506bc4a8aa75a9 -p 257188f80ce1a083e3a88b679b898a7...`
+        # so we will keep the first parent and drop all the others
+        run_command(
+            [
+                "git",
+                "filter-branch",
+                "-f",
+                "--parent-filter",
+                'cut -f 2,3 -d " "',
+                target,
+            ],
+            # git prints nasty warning when filter-branch is used that it's dangerous
+            # this env var prevents it from prints
+            env={"FILTER_BRANCH_SQUELCH_WARNING": "1"},
+        )
+
+    def create_patches(
+        self,
+        git_ref: str,
+        destination: str,
+        files_to_ignore: Optional[List[str]] = None,
+    ) -> List[Tuple[Path, str]]:
+        """
+        Create patches from git commits.
+
+        :param git_ref: start processing commits from this till HEAD
+        :param destination: place the patch files here
+        :return: [(patch_path, msg)] list of created patches (tuple of the file path and commit msg)
+        """
+        contained, git_commit = self.are_child_commits_contained(git_ref)
```

Comment:
`git_commit` doesn't seem to be used anywhere.

Reply:
that's funny, locally it was passing fine